Extension/src/LanguageServer/cppBuildTaskProvider.ts

```diff
@@ -361,6 +361,11 @@ class CustomBuildTaskTerminal implements Pseudoterminal {
         // Do build.
         let command: string = util.resolveVariables(this.command);
         let activeCommand: string = command;
+
+        // Create the exe folder path if it doesn't exists.
+        const exePath: string | undefined = util.resolveVariables(util.findExePathInArgs(this.args));
+        util.createDirIfNotExistsSync(exePath);
+
         this.args.forEach((value, index) => {
```

Comment:
Something I just noticed/realized about this function: We should not be adding spaces or quoting to arguments in an argument array.  It's unnecessary.  We are passing these arguments to `spawn`, which accepts them without quoting/escaping, unless we specify `shell` in the options argument, which we do not need to do.  The adding of `shell` to the options arg should be removed, and we should stop modifying the arguments to add quoting, as it just adds complexity to what could be a simple operation.  In cases where it's necessary to provide a full command line with (quoted) arguments in it, our implementation of normalizeArg is wrong, and should instead be made equivilent to `quote_argument` in GlueWin.cpp and GlueXnix.cpp on the native side.

Reply:
https://nodejs.org/api/child_process.html#child_processspawncommand-args-options

Reply:
Of, if the issue is that we need to invoke using the shell for other reasons (shell script tasks?), then the issue is only that normalizeArgs needs to be updated to match `quote_argument` in GlueWin.cpp and GlueXnix.cpp on the native side.

Reply:
As discussed, we will update `normalizeArgs` in a subsequent PR.